scripts/basic-workflow-tests.sh

```diff
@@ -231,6 +237,18 @@ test_train() {
     fi
 
     ilab model train "${TRAIN_ARGS[@]}"
+
+    if [ "$FULLTRAIN" -eq 1 ]; then
+
+        # TODO Only cuda for now
+        # the train profile specified in test_init overrides the majority of TRAIN_ARGS, including things like num_epochs. While it looks like much of those settings are being lost, they just have different values here.
+        TRAIN_ARGS=("--device=cuda" "--model-path=instructlab/granite-7b-lab" "--data-path=${SCRIPTDIR}/test-data/train_all_pruned_SDG.jsonl" "--quantize-data-type=nf4" "--4-bit-quant")
```

Comment:
Is it right that we override the current `TRAIN_ARGS` array (initialized through `set_defaults`)? Shouldn't this be `TRAIN_ARGS+=(` instead?

Reply:
we want to use the things set in the train profile so we need to get rid of some of the flags already set I believe:

` the train profile specified in test_init overrides the majority of TRAIN_ARGS, including things like num_epochs. While it looks like much of those settings are being lost, they just have different values here.`

Reply:
Ok just wanted to make sure we don't override anything important!